components/script/dom/htmlformelement.rs

```diff
@@ -285,9 +287,21 @@ impl HTMLFormElement {
             FormEncType::UrlEncoded => {
                 let mime: mime::Mime = "application/x-www-form-urlencoded".parse().unwrap();
                 load_data.headers.set(ContentType(mime));
-                serialize(form_data.iter().map(|d| (&*d.name, &*d.value)))
+
+                serialize(form_data.iter().map(|d| (&*d.name, match d.value {
+                    FileOrString::StringData(ref s) => String::from(s.clone()),
+                    FileOrString::FileData(ref f) => String::from(f.name().clone())
+                })))
+            }
+            FormEncType::FormDataEncoded => {
+                let boundary = generate_boundary();
+                let mime = "multipart/formdata; boundary=".to_owned() + &boundary;
```

Comment:
What I mean is that we can [use](http://hyper.rs/mime.rs/mime/enum.TopLevel.html) [these](http://hyper.rs/mime.rs/mime/enum.SubLevel.html) [enums](http://hyper.rs/mime.rs/mime/enum.Attr.html), instead of hardcoding a string.